test/sw_mgmt_spec.rb

```diff
@@ -200,4 +218,40 @@
       Registration::SwMgmt.select_addon_products
     end
   end
+
+  describe ".products_from_repo" do
+    before do
+      expect(Yast::Pkg).to receive(:ResolvableProperties)
+        .and_return(YAML.load_file(fixtures_file("products_legacy_installation.yml")))
+    end
+
+    it "Returns product resolvables from the specified repository" do
+      expect(Registration::SwMgmt.products_from_repo(5)).to have(1).item
+    end
+
+    it "Returns empty list if not product is found" do
+      expect(Registration::SwMgmt.products_from_repo(255)).to be_empty
+    end
+  end
+
+  describe ".select_product_addons" do
+    # just the sle-module-legacy product
+    let(:products) { [YAML.load_file(fixtures_file("products_legacy_installation.yml")).first] }
+
+    it "selects remote addons matching the product resolvables" do
+      available_addons = YAML.load_file(fixtures_file("available_addons.yml"))
+
+      # expect the sle-module-legacy product to be selected
+      expect(available_addons[4]).to receive(:selected)
```

Comment:
NP: this is extremely nitpicking, but it's somehow also related with the idea of testing behavior instead of implementation outlined above.

What about having?
```ruby
  Registration::SwMgmt.select_product_addons(products, available_addons)
  expect(available_addons[4].selected?).to eq true
```

In addition, it will allow you to define the whole thing using contexts to be more declarative... again open to discussion, @jreidinger would not like it :-)

```ruby
describe ".select_product_addons" do
  # just the sle-module-legacy product
  let(:products) { [YAML.load_file(fixtures_file("products_legacy_installation.yml")).first] }
  before do
    Registration::SwMgmt.select_product_addons(products, available_addons)
  end

  context "when there is a remote addon matching the product resolvables" do
    let(:available_addons) { YAML.load_file(fixtures_file("available_addons.yml")) }

    it "selects the addon" do
      expect(available_addons[4].selected?).to eq true
    end

    it "does not select other addons" do
      expect(available_addons[0].selected?).to eq false
    end
  end

  context "when there is no matching remote addon" do
    let(:available_addons) { [] }

    it "reports an error" do
      expect(Yast::Report).to receive(:Error).with(/Cannot find remote product/)
    end
  end
end